gcivil-nyu-org/fall2019-cs-gy-6063-team-debug_entities
# Make test users for Prof & Dan

### Description

* Modified the make_users command to also make test users for Prof & Dan
* Added the make_users command to Procfile so that it'll run on every deployment so that our test users will survive a DB reset
* Added a "forgot password" button to the login page
* Added a template for the password reset page

### Testing Directions

1. Get the code from this PR
1. Delete db.sqlite3
1. Run `python manage.py migrate && python manage.py make_users`
1. See that accounts are made for Prof & Dan, along with the usual test accounts. This simulates what will happen when we deploy after a DB reset.
![image](https://user-images.githubusercontent.com/13837978/69206163-ee666f80-0b19-11ea-814a-6f5fc563c49d.png)
1. Run `python manage.py make_users` again. See that no new users are created because the command realizes that the users already exist. This simulates what will happen when we deploy without a DB reset.
![image](https://user-images.githubusercontent.com/13837978/69206278-3be2dc80-0b1a-11ea-8ff2-63df1c8203e7.png)
1. Sign in as Prof with email gcivil@nyu.edu and password heyhey123. See that it works.
1. Sign in as Dan with email dgopstein@nyu.edu and password heyhey123. See that it works.
1. Log out, click the "forgot password" link on the login page, put in your email address (assuming you already have an account, create one if not), see that you get an email, click the link in it, see that you can reset your password.

### Miscellaneous Comments

* We can't truly test Procfile locally because it only applies to Heroku deployments, and the Heroku environment is different from our local ones. But as noted in the testing steps, we can try to simulate what Procfile will do in various scenarios.

### Checklist
